modules/tokens/utils/types.ts

```diff
@@ -13,13 +13,44 @@ export interface TokenWithSecret extends Token {
 	token: string;
 }
 
-export function tokenFromRow(
+
+export function withoutKeys<T extends object, K extends keyof T>(
+	obj: T,
+	keys: K[]
+): Omit<T, K> {
+	const copy = { ...obj };
+	for (const key of keys) {
+		delete copy[key];
+	}
+	return copy;
+}
+
+export function tokenFromRowWithSecret(
 	row: prisma.Prisma.TokenGetPayload<any>,
+	origToken: string
 ): TokenWithSecret {
 	return {
-		...row,
+		...tokenFromRow(row),
+		token: origToken,
+	}
+}
+
+export function tokenFromRow(
+	row: prisma.Prisma.TokenGetPayload<any>,
+): Token {
+	return {
+		...withoutKeys(row, ["tokenHash"]),
```

Comment:
this is more lines of code and harder to understand than just manually passing in each parameter from the row in to the response.